tls/ttls.c

```diff
@@ -2310,33 +2308,12 @@ ttls_recv(void *tls_data, unsigned char *buf, size_t len, unsigned int *read)
 
 	if (io->msgtype == TTLS_MSG_ALERT) {
 		if (!(r = ttls_handle_alert(tls)))
-			goto skip_record;
+			goto done;
 		return T_DROP;
 	}
 
-	/*
-	 * At this point we have fully prepared data for the upper layer.
-	 * Once we return the data is passed for processing to the upper layer,
-	 * so we reinitialize I/O context for a next message.
-	 */
-	bzero_fast(io->__initoff, sizeof(*io) - offsetof(TlsIOCtx, __initoff));
-
+done:
```

Comment:
It seems we don't need the label any more and can just `return T_OK` instead of `goto done`.

Next, the comments for `T_POSTPONE` and `T_OK` in `tfw_tls_msg_process()` become misleading, since now after the `switch (r) { ... }` in the function we may have an alert of handshake message which should not be passed to upper layer. And this is why you check for `tls->io_in.msgtype == TTLS_MSG_APPLICATION_DATA` - having the old work flow in mind, I couldn't figure this out quickly.

Reply:
> It seems we don't need the label any more and can just `return T_OK` instead of `goto done`.

Replaced `goto done` with `return T_OK`.

> the comments <...> become misleading

Updated the comments so that they now explain the current code.